### PR TITLE
go.mod: update tidb to the new version

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: shirly <AndreMouche@126.com>

### What problem does this PR solve? 

This PR updated `tidb` to the new version since some features of the backoff in `tidb` have updated.

### Check List 

 - Unit test

### Release Note

- No release note

### CODE REVIEW

**Comment by glorv** [comment]
```
[2021-05-31T02:29:12.126Z] panic: runtime error: invalid memory address or nil pointer dereference

[2021-05-31T02:29:12.126Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x2aea092]

[2021-05-31T02:29:12.126Z] 

[2021-05-31T02:29:12.126Z] goroutine 628 [running]:

[2021-05-31T02:29:12.126Z] github.com/pingcap/br/pkg/lightning/backend/kv.(*session).GetInfoSchema(0xc0011a79e0, 0x8, 0x2)

[2021-05-31T02:29:12.127Z] 	<autogenerated>:1 +0x32

[2021-05-31T02:29:12.127Z] github.com/pingcap/tidb/planner/core.rewriteAstExpr(0x3cf19b8, 0xc0011a79e0, 0x3cdb538, 0xc000982fc0, 0xc000e35cc0, 0xc000e5be90, 0x6, 0x6, 0x350b6e0, 0x37815ff, ...)

[2021-05-31T02:29:12.127Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20210527074735-a5400cc602d1/planner/core/expression_rewriter.go:68 +0x4c

[2021-05-31T02:29:12.127Z] github.com/pingcap/br/pkg/lightning/backend/kv.collectGeneratedColumns(0xc0011a79e0, 0xc000dc8480, 0xc000846db0, 0x6, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0)

[2021-05-31T02:29:12.127Z] 
```

@AndreMouche  Seems tidb has introduced a new method `GetInfoSchema` for `sessionctx.Context`, we maybe need to also implement it in `kv.session`

File: cmd/br/cmd.go [hunk 1]
```diff
@@ -154,8 +153,6 @@ func Init(cmd *cobra.Command) (err error) {
 		if len(slowLogFilename) != 0 {
 			tidbLogCfg.SlowQueryFile = slowLogFilename
 		} else {
-			// Hack! Discard slow log by setting log level to PanicLevel
-			logutil.SlowQueryLogger.SetLevel(logrus.PanicLevel)
```
Comment (AndreMouche) [inline cmd/br/cmd.go:158]: We remove this Line because L161 already include it in the tidb's new verion

**Review by kennytm — APPROVED** [review]
/lgtm

**Comment by AndreMouche** [comment]
It seems something wrong in `idc-jenkins-ci-br/integration_test`? @glorv @zhouqiang-cl

**Comment by glorv** [comment]
```
[2021-06-03T03:21:18.498Z] TEST: [br_full_index] fail on fast checksum: required 3 databases checked, but only 0 dbs checked
```

@3pointer PTAL

**Comment by kennytm** [comment]
@glorv the error message is because the log file somehow does not exist. 

```sh
checksum_count=$(cat $LOG | grep "checksum success" | wc -l | xargs)

if [ "${checksum_count}" -lt "$DB_COUNT" ];then
    echo "TEST: [$TEST_NAME] fail on fast checksum: required $DB_COUNT databases checked, but only ${checksum_count} dbs checked"
    echo $(cat $LOG | grep checksum)
    exit 1
fi
```

**Review by glorv — APPROVED** [review]

**Pull request merged**